data/collector_test.go

```diff
@@ -191,59 +202,24 @@ func TestClose(t *testing.T) {
 
 	// Close and measure fileSize.
 	c.Close()
-	fileSize := target1.Size()
+	files := getAllFiles(target.Directory)
+	test.That(t, len(files), test.ShouldEqual, 1)
+	fileSize1 := files[0].Size()
 
 	// Assert capture is no longer being called/file is no longer being written to.
 	time.Sleep(time.Millisecond * 25)
-	test.That(t, target1.Size(), test.ShouldEqual, fileSize)
-}
-
-func TestSetTarget(t *testing.T) {
-	l := golog.NewTestLogger(t)
-	tmpDir := os.TempDir()
-	md1 := v1.DataCaptureMetadata{
-		ComponentName: "someFirstThing",
-	}
-	md2 := v1.DataCaptureMetadata{
-		ComponentName: "someSecondThing",
-	}
-	target1, _ := datacapture.NewFile(tmpDir, &md1)
-	target2, _ := datacapture.NewFile(tmpDir, &md2)
-	defer os.Remove(target1.GetPath())
-	defer os.Remove(target2.GetPath())
-
-	params := CollectorParams{
-		ComponentName: "testComponent",
-		Interval:      time.Millisecond * 15,
-		MethodParams:  map[string]*anypb.Any{"name": fakeVal},
-		Target:        target1,
-		QueueSize:     queueSize,
-		BufferSize:    bufferSize,
-		Logger:        l,
-	}
-	c, _ := NewCollector(dummyStructCapturer, params)
-	c.Collect()
-	time.Sleep(time.Millisecond * 30)
-
-	// Change target, verify that target1 was written to.
-	c.SetTarget(target2)
-	sizeTgt1 := target1.Size()
-	test.That(t, target1.Size(), test.ShouldBeGreaterThan, 0)
-
-	// Verify that tgt2 was written to, and that target1 was not written to after the target was changed.
-	time.Sleep(time.Millisecond * 30)
-	c.Close()
-	test.That(t, target1.Size(), test.ShouldEqual, sizeTgt1)
-	test.That(t, target2.Size(), test.ShouldBeGreaterThan, 0)
+	filesAfterWait := getAllFiles(target.Directory)
+	test.That(t, len(filesAfterWait), test.ShouldEqual, 1)
+	test.That(t, filesAfterWait[0].Size(), test.ShouldEqual, fileSize1)
 }
 
 // TestCtxCancelledLoggedAsDebug verifies that context cancelled errors are logged as debug level instead of as errors.
 func TestCtxCancelledLoggedAsDebug(t *testing.T) {
 	logger, logs := golog.NewObservedTestLogger(t)
+	// TODO: replace all usages of os.TempDir with proper os/io temp dir method
```

Comment:
Could you add a low pri backlog ticket for this? Helpful to have associated tickets for even tiny todos, e.g. if we have a bug cleanup week.

Reply:
Went ahead and just did this, replacing with `t.TempDir()` which auto cleans things up after each test, too